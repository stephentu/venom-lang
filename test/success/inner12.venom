def Foo(x::int) -> int =

  class Bar
    def Baz() = 
      def Quix() = 
        print("HERE 1");
      end
      if x > 10 then
        x = x + 1;
        Quix();
      else
        def AAA() = 
          print("HERE 2");
        end
        AAA();
        x = x + 2;
      end
    end
  end

  def mkBar() -> Bar = Bar(); end

  b = mkBar();
  b.Baz();

  return x;
end

x0 = Foo(0);
x1 = Foo(100);
print(x0);
print(x1);
