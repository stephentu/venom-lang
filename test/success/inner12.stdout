HERE 2
HERE 1
2
101
